Plugins/Json/src/JsonGeometryConverter.cpp

```diff
@@ -611,17 +611,10 @@ json Acts::JsonGeometryConverter::surfaceMaterialToJson(
   json smj;
 
   // lemma 0 : accept the surface
-  auto convertMaterialProperties =
+  auto encodeMaterialProperties =
       [](const Acts::MaterialProperties& mp) -> std::vector<float> {
-    // convert when ready
-    if (mp) {
-      /// Return the thickness in mm
-      return {
-          mp.material().X0(), mp.material().L0(),          mp.material().Ar(),
-          mp.material().Z(),  mp.material().massDensity(), mp.thickness(),
-      };
-    }
-    return {};
+    const auto cn = mp.material().parameters();
+    return {cn[0], cn[1], cn[2], cn[3], cn[4], mp.thickness()};
```

Comment:
Here if(mp.material() == Acts::Material()) you should return {}. This would solve our issue.

Reply:
What do we do for finite vacuum? `[nullptr, thickness]`?

Reply:
Yes, or we just write the thickness and add a if (size == 1) to then jsonToMaterialMatrix.

Reply:
I added a proposal for updating the encoding/decoding to avoid any ambiguity.

Indefinite material is encoded either as a JSON array `[1, 2, 3, 4, 5]` or directly as `null` in case of vacuum. A concrete material slab is always encoded as a JSON object with two members
```
{
  "material": [1, 2, 3, 4, 5], // or null in case of vacuum
  "thickness": 1.2,
}
```
This allows encoding of all special cases (vacuum w/ zero thickness, vacuum w/ non-zero thickness, real material w/ zero thickness) without loss of information at the expense of a slightly more verbose output.

This way the `Material` encoding is the same in both cases and indefinite and concrete materials are clearly encoded in two different ways.